main.tf

```diff
@@ -36,36 +36,62 @@ data "aws_iam_policy_document" "trusted_policy" {
   }
 }
 resource "aws_iam_role_policy" "role_policy" {
-  name = "IT-Management-Cloud-Integration-Role-Policy"
-  role = aws_iam_role.this.id
-  policy = data.aws_iam_policy_document.role_policy.json
+  name   = "Admina-Integration-Role-Policy"
+  role   = aws_iam_role.this.id
+  policy = var.read_only ? data.aws_iam_policy_document.read_only_role_policy.json : data.aws_iam_policy_document.role_policy.json
 }
 
 data "aws_iam_policy_document" "role_policy" {
   statement {
-    sid = "IntegrationAccessPolicy"
+    sid    = "IntegrationAccessPolicy"
     effect = "Allow"
     actions = [
-                "iam:GetAccessKeyLastUsed",
-                "iam:ListAccessKeys",
-                "iam:ListAttachedUserPolicies",
-                "iam:ListGroupsForUser",
-                "iam:ListMFADevices",
-                "iam:ListRolePolicies",
-                "iam:ListRoles",
-                "iam:ListUsers",
-                "iam:ListUserTags",
-                "iam:DeleteUser",
-                "iam:DeleteAccessKey",
-                "iam:DeleteRole",
-                "iam:ListAttachedRolePolicies",
-                "iam:DetachRolePolicy",
-                "iam:DeleteLoginProfile",
-                "iam:ListAttachedUserPolicies",
-                "iam:DetachUserPolicy",
-                "account:GetContactInformation"
+      "iam:GetAccessKeyLastUsed",
+      "iam:ListAccessKeys",
+      "iam:ListAttachedUserPolicies",
+      "iam:ListGroupsForUser",
+      "iam:ListMFADevices",
+      "iam:ListRolePolicies",
+      "iam:ListRoles",
+      "iam:ListUsers",
+      "iam:ListUserTags",
+      "iam:DeleteUser",
+      "iam:DeleteAccessKey",
+      "iam:DeleteRole",
+      "iam:ListAttachedRolePolicies",
+      "iam:DetachRolePolicy",
+      "iam:DeleteLoginProfile",
+      "iam:ListAttachedUserPolicies",
+      "iam:DetachUserPolicy",
+      "iam:CreateUser",
```

Comment:
This looks a bit strange, other permissions are read/delete related, is this for provisioning? But we can't attach anything to the user 🤔

Reply:
@narxeba Can you confirm it ?

Reply:
Yep, this is for provisioning.
>But we can't attach anything to the user 🤔

Hmm, I see. Let me check the current implementation.

Reply:
@narxeba How about this ? I would like to proceed this PR

Reply:
@umisora So right now our provisioning for IAM can only add a new user without any roles, permissions :wwww: You can process this PR. I'll work on improving provisioning for IAM later